source/module_io/cube_io.h

```diff
@@ -8,45 +8,91 @@
 
 namespace ModuleIO
 {
-bool read_cube(
+extern bool read_cube(
 #ifdef __MPI
-    Parallel_Grid* Pgrid,
+    const Parallel_Grid*const Pgrid,
 #endif
-    int my_rank,
-    std::string esolver_type,
-    int rank_in_stogroup,
-    const int& is,
+    const int my_rank,
+    const std::string esolver_type,
+    const int rank_in_stogroup,
+    const int is,
     std::ofstream& ofs_running,
-    const int& nspin,
+    const int nspin,
     const std::string& fn,
-    double* data,
-    const int& nx,
-    const int& ny,
-    const int& nz,
+    double*const data,
+    const int nx,
+    const int ny,
+    const int nz,
     double& ef,
-    const UnitCell* ucell,
+    const UnitCell*const ucell,
     int& prenspin,
-    const bool& warning_flag = true);
+    const bool warning_flag = true);
 
-void write_cube(
+extern void write_cube(
 #ifdef __MPI
-    const int& bz,
-    const int& nbz,
-    const int& nplane,
-    const int& startz_current,
+    const int bz,
+    const int nbz,
+    const int nplane,
+    const int startz_current,
 #endif
-    const double* data,
-    const int& is,
-    const int& nspin,
-    const int& iter,
+    const double*const data,
+    const int is,
+    const int nspin,
+    const int iter,
     const std::string& fn,
-    const int& nx,
-    const int& ny,
-    const int& nz,
-    const double& ef,
-    const UnitCell* ucell,
-    const int& precision = 11,
-    const int& out_fermi = 1); // mohan add 2007-10-17
+    const int nx,
+    const int ny,
+    const int nz,
+    const double ef,
+    const UnitCell*const ucell,
+    const int precision = 11,
+    const int out_fermi = 1); // mohan add 2007-10-17
+
+
+// when MPI:
+//      read file as order (ixy,iz) to data[ixy*nz+iz]
+// when serial:
+//      read file as order (ixy,iz) to data[iz*nxy+ixy]
+extern void read_cube_core_match(
```

Comment:
could you add some annotation to describe what this function for? I can hardly understand why reading the cube file depends on whether the core match or mismatch...